### PR TITLE
Add load balancer deployment support

### PR DESCRIPTION
**Pull request by mkolesnik**
This makes it easy to deploy a test environment with the load balancer, by calling e.g. `make deploy using=load-balancer`.

For kind based deployments, `metallb` is deployed and used.

Depends on #952 

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: release-notes/20220919-load-balancer.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<!-- markdownlint-disable MD041 -->
+Load balancer support was added to easily deploy with Submariner with the load balancer in front.
+To use, simply use the target (e.g. `deploy`) with `USING=load-balancer` or `LOAD_BALANCER=true`.
+For kind based deployments, [metallb](https://metallb.universe.tf/) is deployed to provide the capability.
+The version can be specified using `METALLB_VERSION=x.y.z`.
```
Comment (nyechiel) [inline release-notes/20220919-load-balancer.md]:
```suggestion
Support was added in the Shipyard project to easily deploy Submariner with a LoadBalancer type Service in front.
To use, simply specify the target (e.g. `deploy`) with `USING=load-balancer` or `LOAD_BALANCER=true`.
For kind-based deployments, [MetalLB](https://metallb.universe.tf/) is deployed to provide the capability.
The MetalLB version can be specified using `METALLB_VERSION=x.y.z`.
```

File: scripts/shared/lib/clusters_kind [hunk 2]
```diff
@@ -203,6 +208,32 @@ function deploy_ovn_cni(){
     echo "OVN CNI deployed."
 }
 
+function deploy_load_balancer() {
+    local kind_ip start_ip end_ip
```
Comment (yboaron) [inline scripts/shared/lib/clusters_kind]: If LB is deployed on 2 clusters , IIUC both clusters will use the same IPAM range x1.x2.x3.100 - x1.x2.x3.250

We should have separate range for each cluster

Comment (mkolesnik) [reply scripts/shared/lib/clusters_kind]: Yes, thats why cluster CIDR is used in the IP range, though now that I think of it probably better to use the GN CIDR..

Comment (mkolesnik) [reply scripts/shared/lib/clusters_kind]: I updated so that GN CIDR is used when GN enabled, but generally your concern has already been addressed.

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

**Comment by nyechiel** [comment]
@mkolesnik I will let you merge this; can you also update the website docs?

**Pull request merged**